preprocessor/transcriptions/episode_info_processor.py

```diff
@@ -1,69 +1,113 @@
 import json
 from pathlib import Path
 import re
-from typing import Optional
+from typing import (
+    Optional,
+    Tuple,
+)
 
 from preprocessor.utils.error_handling_logger import ErrorHandlingLogger
 
 
 class EpisodeInfoProcessor:
-    def __init__(self, jsons_dir: Path, episodes_info_json: Path, output_path: Path, logger: ErrorHandlingLogger):
+    def __init__(
+        self,
+        jsons_dir: Path,
+        episodes_info_json: Path,
+        output_path: Path,
+        logger: ErrorHandlingLogger,
+        series_name: str = "",
+    ):
         self.__jsons_dir: Path = jsons_dir
-
         with open(episodes_info_json, "r", encoding="utf-8") as f:
             self.__episodes_info: json = json.load(f)
-
         self.__output_path: Path = output_path
         self.__logger: ErrorHandlingLogger = logger
-
+        if not series_name:
+            series_name = self.__output_path.parent.name.lower()
```

Comment:
to moze dopisz logger warning ze nie podano name i dedukujesz z folderu i co wydedukowales